### PR TITLE
Convert hydrate.js to ES5 for IE11 compatibility (closes #77)

### LINKED ISSUE #77: Does not work in IE11 as hydrate.js is not transpiled
In IE11, an exception occurs in every page that uses `next-mdx-remote/hydrate`. This is because `hydrate.js` is not transpiled.

As a workaround, we can use https://github.com/martpie/next-transpile-modules with the following `next.config.js`:

```js
const withTM = require('next-transpile-modules')(['next-mdx-remote/hydrate']);

module.exports = withTM();
```

But ideally, the source should be transpiled before publishing to npm.

### PR DESCRIPTION
**Pull request by cknitt**

### CODE REVIEW

File: hydrate.js [hunk 1]
```diff
@@ -27,32 +32,34 @@ module.exports = function hydrate(
...
+  React.useEffect(function () {
+    var handle = window.requestIdleCallback(function () {
       // first we set up the scope which has to include the mdx custom
       // create element function as well as any components we're using
-      const fullScope = { mdx, ...components, ...scope }
-      const keys = Object.keys(fullScope)
-      const values = Object.values(fullScope)
+      var fullScope = Object.assign({ mdx: MDX.mdx }, components, scope);
+      var keys = Object.keys(fullScope)
+      var values = Object.values(fullScope)
```
Comment (jescalan) [inline hydrate.js:41]: IE11 doesn't support `Object.values` sadly: https://caniuse.com/object-values

File: hydrate.js [hunk 2]
```diff
@@ -27,32 +32,34 @@ module.exports = function hydrate(
...
       // and all our components in scope for the function, which is the case here
       // we pass the names (via keys) in as the function's args, and execute the
       // function with the actual values.
-      const hydratedFn = new Function(
-        'React',
-        ...keys,
-        `${compiledSource}
-      return React.createElement(MDXContent, {});`
-      )(React, ...values)
+      var hydrateFn = Reflect.construct(
```
Comment (jescalan) [inline hydrate.js:48]: Not supported by IE11: https://caniuse.com/mdn-javascript_builtins_reflect

**Review by jescalan — CHANGES_REQUESTED** [review]
Thank you for this good work! There are still a few spots where IE11 would fail. Did you test this in IE yet?

**Comment by cknitt** [comment]
Yes, I tested this successfully in IE11.

The unsupported functions are automatically polyfilled by Next.js, see https://nextjs.org/docs/basic-features/supported-browsers-features#polyfills (which unfortunately does not list the exact polyfill that are applied, but `Object.assign` and `Reflect.construct` are definitely among them).

**Comment by jescalan** [comment]
Oh that's great! hadn't considered this. Thank you! I'll try to get this shipped out soon 😀

**Comment by cknitt** [comment]
Great, thanks a lot! 😀

**Review by jescalan — APPROVED** [review]

**Pull request merged**